virtual_ecosystem/models/plants/plants_model.py

```diff
@@ -651,7 +666,38 @@ def calculate_turnover(self) -> None:
         self.data["root_turnover_c_p_ratio"] = xr.full_like(
             self.data["elevation"], 656.7
         )
-
         self.data["nitrogen_fixation_carbon_supply"] = xr.full_like(
             self.data["elevation"], 0.01
         )
+        self.data["root_carbohydrate_exudation"] = xr.full_like(
+            self.data["elevation"], 0.025
+        )
+
+    def calculate_nutrient_uptake(self) -> None:
+        """Calculate uptake of soil nutrients by the plant community.
+
+        This function calculates the rate a which plants take up inorganic nutrients
+        (ammonium, nitrate, and labile phosphorus) from the soil.
+
+        Warning:
+            At present, this function just calculates uptake based on an entirely made
+            up function, and does not link to plant dynamics in any way.
+        """
+
+        # Assume plants can take 0.1% of the available nutrient per day, uptake stops
+        # (rather than goes negative) if the soil nutrient concentrations go negative
+        self.data["plant_ammonium_uptake"] = xr.where(
```

Comment:
Noob question about python: when I googled `xr.where` I got to the page `xarray.where`. Does python functions always have a shortform counterpart?

Reply:
If you look at the top of the file, you'll see `import xarray as xr`. This then allows us to use the short name within that code file. 

Some packages are commonly used this way (`numpy` -> `np`, `xarray` -> `xr` etc) and others just aren't particularly. The abbreviations tend to be fairly universal, but there isn't actual anything to stop someone doing `import xarray as I_really_like_typing_so_this_is_xarray`

Reply:
There's some `python` packages that have standard condensed names (I can only think of `np` for `numpy` and `xr` for `xarray` off the top of my head).

These get defined at the top of the script using the following

```python
import xarray as xr
```

Reply:
SNAP! ♣️ ♠️ ❤️ ♦️